cmd/main.go

```diff
@@ -27,13 +25,20 @@ const (
 func main() {
 	var command *cobra.Command
 
+	o := cli.ArgoCDCLIOptions{
+		PluginHandler: cli.NewDefaultPluginHandler([]string{"argocd"}),
+		Arguments:     os.Args,
+	}
+
 	binaryName := filepath.Base(os.Args[0])
 	if val := os.Getenv(binaryNameEnv); val != "" {
 		binaryName = val
 	}
+	o.Arguments[0] = binaryName
+
 	switch binaryName {
 	case "argocd", "argocd-linux-amd64", "argocd-darwin-amd64", "argocd-windows-amd64.exe":
-		command = cli.NewCommand()
+		command = cli.NewDefaultArgoCDCommandWithArgs(o)
```

Comment:
This is a bit too invasive. I think plugins should be implemented as a simple cobra subcommand. See my other comment about defining a dedicated subcommand for running plugins.

Reply:
Sorry if I didn't explain it clearly. My suggestion is not have every single plugin to implemented as subcommands. It is just to create a single subcommand called `plugin` so that can be dedicated to finding binaries in the PATH and executing (example: `argocd plugin <some-plugin>`). This is mainly to avoid the loop to decide if the invoked command is a plugin or not.